batch/batch/instance_config.py

```diff
@@ -106,13 +110,19 @@ def cost_per_hour_from_cores(
         utilized_cores_mcpu: int,
     ) -> float:
         assert 0 <= utilized_cores_mcpu <= self.cores * 1000
-        memory_in_bytes = self.cores_mcpu_to_memory_bytes(utilized_cores_mcpu)
+        if self.job_private:
+            memory_in_bytes = self.instance_memory()
```

Comment:
This seems fine to me, you could further make your point here by doing `assert utilized_cores_mcpu == self.cores * 1000`.